### PR TITLE
CDPTP-195 Set payment uplift flags on new EarlyCareerTeacherProfile records

### PR DESCRIPTION
**Pull request by blackrat**
### Context

Track and Pay require flags against the Early Career Teacher Profiles to ensure that we are picking up the participants
to whom uplift is required. The decision about the flags boils down to the fact that we don't care if the underlying information changes, we just care if it was applicable at the point of the registration.

This is to ensure that when a new registration happens, we set those flags against the early career teacher.

### Changes proposed in this pull request

Add code to pick up the current uplift flags from sparsity and pupil premium and set them in the early career teacher profile.

### Guidance to review

Would rather have this in a service or other separated section, but there's been some work on building a multistep form that I don't want to refactor at the moment to make this more separated, so it's a callback for now.

### Testing

New early_career_teacher_profile records will have the appropriate flags set.

### Review Checks
- [ ] All pages have automated accessibility checks via cypress
- [ ] All pages have visual tests via cypress + percy
- [ ] All `School` queries are correctly scoped - `eligible` when they need to be

### How can I view this in a review app?

No visible changes.

### CODE REVIEW

File: app/models/participant_declaration.rb [hunk 1]
```diff
@@ -4,18 +4,19 @@ class ParticipantDeclaration < ApplicationRecord
...
   scope :declared_as_between, ->(start_date, end_date) { where(declaration_date: start_date..end_date) }
   scope :submitted_between, ->(start_date, end_date) { where(created_at: start_date..end_date) }
-  scope :active_uplift_for_lead_provider, lambda { |lead_provider|
-    active_for_lead_provider(lead_provider).joins(:early_career_teacher_profile)
-      .where("early_career_teacher_profiles.sparsity_uplift OR early_career_teacher_profiles.pupil_premium_uplift")
-  }
+
+  # Declaration aggregation scopes
+  scope :count_active_for_lead_provider, ->(lead_provider) { active_for_lead_provider(lead_provider).count }
+  scope :count_active_uplift_for_lead_provider, ->(lead_provider) { active_for_lead_provider(lead_provider).uplift.count }
```
Comment (plisovin) [inline app/models/participant_declaration.rb:21]: Looks much cleaner now

File: spec/services/early_career_teachers/create_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+RSpec.describe EarlyCareerTeachers::Create do
+  let(:user) { create :user }
+  let(:school) { create :school }
+  let(:pupil_premium_school) { create :school, :pupil_premium_uplift }
+  let(:sparsity_school) { create :school, :sparsity_uplift }
+  let(:uplift_school) { create :school, :sparsity_uplift, :pupil_premium_uplift }
+  let(:cohort) { create :cohort, :current }
+  let(:mentor_profile) { create :mentor_profile }
```
Comment (kicferk1) [inline spec/services/early_career_teachers/create_spec.rb:10]: Should we test that mentor profile is set on ect profile somewhere?

File: spec/services/early_career_teachers/create_spec.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  it "creates an Early Career Teacher Profile record" do
+    before_count = EarlyCareerTeacherProfile.count
+    early_career_teacher_profile = described_class.call(email: user.email, full_name: user.full_name, school_id: school.id, cohort_id: cohort.id, mentor_profile_id: mentor_profile.id)
+    after_count = EarlyCareerTeacherProfile.count
+    expect(after_count).to eq(before_count + 1)
```
Comment (kicferk1) [inline spec/services/early_career_teachers/create_spec.rb:16]: I would make this its own test, without checking anything else. And use the form of 
```
    expect {
      described_class.call(email: user.email, full_name: user.full_name, school_id: school.id, cohort_id: cohort.id, mentor_profile_id: mentor_profile.id)
    }.to change { EarlyCareerTeacherProfile.count }.by(1)
```

Also, should we add a test for the edge case when user does not exist? And the test could say that 'when the user does not exist, it creates a new user'.

**Review by ethanmills — APPROVED** [review]
Happy with this once the test comments are addressed

**Review by kicferk1 — APPROVED** [review]
